### ISSUE TITLE
[FEATURE] - Add before and after hooks to models

### ISSUE THREAD

**Issue by FacundoSpira** [premieroctet/next-admin#386]
### Summary

Right now if we need to manipulate the data before submitting it, we need to do it in the global route, by having a bunch of if-else branches depending on the `context.params.nextadmin` array. Same thing happens with after actions.

It would be really useful to be able to define `before` and `after` functions in the options config at a specific model edit level

### Basic Example

One particular use case that I'm thinking about is creating a User. Usually we want to:
- Hash the password -> this should happen before adding the record to the database
- Send an email to the user -> this should happen after the record is added to the database. 

I can imagine something like this:
```ts
const options = {
  model: {
    User: {
      edit: {
        before: (request, context) => {
          // manage the request form data here to add the hashed password
          return request;
        },
        after: (response) => {
          // send email
          return response;
        }
      }
    }
  }
};
```

This would be pretty similar to how Admin.js manages it via what they call [actions](https://docs.adminjs.co/basics/action#using-before-and-after-hooks)

### Drawbacks

Can't see any drawback.

**Comment by FacundoSpira** [comment]
Hey @foyarash how are you doing?
Is there any estimation on when this PR, alongside the other open PRs will be merged and released?
Really looking forward to start testing and using the new features 🚀

**Comment by foyarash** [comment]
Hello @FacundoSpira , we will try merging the PRs today, aswell as the release for those PRs

**Comment by foyarash** [comment]
Available in v5.3.0
